youtube_dl/downloader/hls.py

```diff
@@ -167,8 +168,11 @@ def is_ad_fragment_end(s):
                             'giving up after %s fragment retries' % fragment_retries)
                         return False
                     if decrypt_info['METHOD'] == 'AES-128':
+                        info_key = info_dict.get('hls_aes128_key')
```

Comment:
I introduce a new internal way of passing HLS keys, in case the native HLS implementation (this file) is unable to correctly extract it. In this specific case, this is because of a cookie issue described in a source comment – see below.